### ISSUE TITLE
[BUG] Queue on playlist is maxed at 1000 or 501 songs.

### ISSUE THREAD

**Issue by StubbyBook7** [namidaco/namida#245]
Hello, this is my first time posting an issue so sorry ahead of time if this is in the wrong section, but I do believe that this is a bug. Thank you for reading this and thank you for your hard work.

### What Happened

When pressing shuffle or selecting a soundtrack on a playlist of more than 1001 soundtracks, the queue will only start with 1000 songs or in the case of selecting a soundtrack, only with 501 soundtracks. This also goes for the overall soundtracks, favorites tab, history, anything that can start a queue.

## Expected Behavior

What should have happened is that when the shuffle or a soundtrack is selected, the queue should include all soundtracks from the playlist to the queue.

## Current Behavior
 What happens instead is that in a playlist of more than 1001 soundtracks, when the shuffle is selected it will cut off the rest of the soundtracks and only start with 1000 or if a soundtrack is selected it will cut off the rest again and only choose the first 501 soundtracks to the queue.

## Steps to Reproduce

1. Create playlist with 1001 soundtracks or have an overall soundtracks of the same quantity.
2. Select shuffle, and when the queue appears it will only show 1000 songs or select one song from the playlist and the queue will only start with 501 songs
3. This can be done with any tab that can start a queue such as the favorites tab, history, most played, as long as it contains more than 1001 songs.

## Screenshots

![Screenshot_20240627_034343](https://github.com/namidaco/namida/assets/73491843/da4a0a43-0d68-460f-9e79-5ce610ab1c87)
![Screenshot_20240627_034336](https://github.com/namidaco/namida/assets/73491843/17d68172-fa86-4eb2-be7c-8b05f2187bf3)
![Screenshot_20240627_034305](https://github.com/namidaco/namida/assets/73491843/a5cfd5c8-dd0f-4aab-a8f6-3136d7ac73c7)

Also here is a video showing an example of the problem (I tried uploading through here but there is file size limit for videos so I did it through YouTube instead): https://youtu.be/RG_gs0wGUXk?si=725VTVYFR9PG7SNF

## Logs

[logs_2.5.6_240522118.txt](https://github.com/user-attachments/files/16009801/logs_2.5.6_240522118.txt)
[logs_tagger_2.5.6_240522118.txt](https://github.com/user-attachments/files/16009802/logs_tagger_2.5.6_240522118.txt)
[logs_clean_2.5.6_240522118.txt](https://github.com/user-attachments/files/16009803/logs_clean_2.5.6_240522118.txt)

## Possible Solution
 
Unfortunately I do not know how to code... that is all.

**Comment by lewismalili** [comment]
Wow that's a detailed analysis. I'm also facing a similar bug when using the app. And also when I remove songs from a playlist there are blank spaces that remain in the place where the track was located. The playlist doesn't re-order itself automatically. It takes a couple of seconds before the gaps get closed.

**Comment by StubbyBook7** [comment]
> Wow that's a detailed analysis. I'm also facing a similar bug when using the app. And also when I remove songs from a playlist there are blank spaces that remain in the place where the track was located. The playlist doesn't re-order itself automatically. It takes a couple of seconds before the gaps get closed.

Yeah I noticed that too, but this issue seems to have been taken care of (I think) it was on #219, found it while perusing to make sure I wasn't making a duplicate post, it just hasn't been implemented yet.

**Comment by MSOB7YY** [comment]
that's really a duplicate of #31, its not a bug but rather a limitation to avoid performance issues, as explained here https://github.com/namidaco/namida/issues/31#issuecomment-1770926250, u can bypass this behavior by playing 1 track, selecting all others and play next/last. 

about the blank area when removing track from queue (or playlist @lewismalili), thats fixed and available in beta (ref #219, thanks @StubbyBook7).


quite the detailed issue tho thank u for that, let me know if ya guys don't agree with something here, ty

**Comment by StubbyBook7** [comment]
Ah crap, sorry! I guess I didn't look for the post properly and accidentally made a duplicate post (then again made this post at 3 in the morning so I wasn't fully paying attention), I went ahead and tried your method that you recommended https://github.com/namidaco/namida/issues/31#issuecomment-1773765879 and it worked great! I didn't know the limit was an actual limitation so I just assumed it was a bug or something like that, so thank you for clarifying the situation!

**Comment by MSOB7YY** [comment]
nahhh thank u for the awesome detailed report and happy u understand the situation <3 gonna close this then ^^

**Comment by MSOB7YY** [comment]
so uh update to this, i allowed unlimited queue if u play using any button (except when pressing the track tile itself)
can be checked in v4.8.3 beta here https://github.com/namidaco/namida-snapshots/releases
